tests/e2e-kubernetes/testsuites/util.go

```diff
@@ -76,6 +78,36 @@ func checkListingPath(f *framework.Framework, pod *v1.Pod, path string) {
 	e2evolume.VerifyExecInPodSucceed(f, pod, fmt.Sprintf("ls %s", path))
 }
 
+func checkListingPathWithEntries(f *framework.Framework, pod *v1.Pod, path string, entries []string) {
+	cmd := fmt.Sprintf("ls %s", path)
+	stdout, stderr, err := e2evolume.PodExec(f, pod, cmd)
+	framework.ExpectNoError(err,
+		"%q should succeed, but failed with error message %q\nstdout: %s\nstderr: %s",
+		cmd, err, stdout, stderr)
+
+	gomega.Expect(strings.Fields(stdout)).To(gomega.Equal(entries))
+}
+
+// checkBasicFileOperations verifies basic file operations works in the given `basePath` inside the `pod`.
+func checkBasicFileOperations(f *framework.Framework, pod *v1.Pod, basePath string) {
+	framework.Logf("Checking basic file operations inside pod %s at %s", pod.UID, basePath)
+
+	dir := filepath.Join(basePath, "test-dir")
+	first := filepath.Join(basePath, "first")
+	second := filepath.Join(dir, "second")
+
+	seed := time.Now().UTC().UnixNano()
+	testWriteSize := 1024 // 1KB
+
+	checkWriteToPath(f, pod, first, testWriteSize, seed)
+	checkReadFromPath(f, pod, first, testWriteSize, seed)
```

Comment:
Can we do multiple read checks for verifying it works a second time (cache works)?

Reply:
Addressed with https://github.com/awslabs/mountpoint-s3-csi-driver/pull/292/commits/7b15d44b50b9600ef4767fbea776378bdedff1bc